src/core/lib/iomgr/tcp_server_posix.c

```diff
@@ -736,18 +745,21 @@ void grpc_tcp_server_shutdown_starting_add(grpc_tcp_server *s,
 
 void grpc_tcp_server_unref(grpc_exec_ctx *exec_ctx, grpc_tcp_server *s) {
   if (gpr_unref(&s->refs)) {
-    /* Complete shutdown_starting work before destroying. */
     grpc_exec_ctx local_exec_ctx = GRPC_EXEC_CTX_INIT;
+    bool finish_ctx = false;
+    /* FIXME: API allows a NULL exec_ctx, although this might cause us to delete
```

Comment:
I removed uses of NULL exec_ctx and changed the API to always require non-NULL exec_ctx.

The tests look flaky, but let me know if there are any I should fix.